crates/rolldown_plugin/src/types/hook_resolve_id_output.rs

```diff
@@ -8,3 +8,14 @@ pub struct HookResolveIdOutput {
   pub normalize_external_id: Option<bool>,
   pub side_effects: Option<HookSideEffects>,
 }
+
+impl From<ResolvedId> for HookResolveIdOutput {
```

Comment:
I don't really recommend this kind of implicit converting.

Could you add something like

```rust
impl ResolvedId {
  // This makes us easily return the result of `ctx.resolve()` as return value of the `resolve_id` hook.
  fn into_hook_resolve_id_output() {}
}
```